app/src/main/java/com/jin/jjinweather/layer/ui/onboarding/OnboardingScreen.kt

```diff
@@ -8,25 +9,65 @@ import androidx.compose.material3.Button
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.unit.sp
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.rememberPermissionState
+import com.jin.jjinweather.layer.domain.model.PermissionState
 
+@OptIn(ExperimentalPermissionsApi::class)
 @Composable
 fun OnboardingScreen(viewModel: OnboardingViewModel, onNavigateToTemperature: () -> Unit) {
-    Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
+    val locationGrantedStatus by viewModel.locationPermissionState.collectAsState()
+    val composePermissionState = rememberPermissionState(
+        permission = Manifest.permission.ACCESS_COARSE_LOCATION
+    )
+    val weatherState by viewModel.weatherState.collectAsState()
+
+    // 권한 상태 업데이트
+    LaunchedEffect(composePermissionState.status) {
+        viewModel.updateLocationPermissionStatus(composePermissionState.status)
+    }
+
+    // 위치 권한이 승인되었을 때 날씨 로드
+    LaunchedEffect(locationGrantedStatus) {
+        if (locationGrantedStatus == PermissionState.GRANTED) {
+            viewModel.loadWeather()
+        }
+    }
+
 
+    Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
         Box(
             modifier = Modifier
                 .fillMaxSize()
                 .padding(innerPadding), contentAlignment = Alignment.Center
         ) {
+            // todo : tutorial pager
             Column {
-                // todo : tutorial pager
-                //        permission
-                Text("Tutorial Screen", fontSize = 24.sp)
-                Button(onClick = onNavigateToTemperature) { Text("GO TO MAIN") }
+                when (locationGrantedStatus) {
+                    PermissionState.GRANTED -> {
+                        WeatherContentUI(weatherState, onNavigateToTemperature)
+                    }
+
+                    else -> {
+                        PermissionRequestUi { composePermissionState.launchPermissionRequest() }
+                    }
+                }
```

Comment:
https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements

코틀린 코딩 컨벤션 문서를 거의 외울정도가 되어야 하는데, 아직 읽으신적이 없으신것 같아요.
불필요한 중괄호는 넣지 않으셔야 해요.
```suggestion
                when (locationGrantedStatus) {
                    PermissionState.GRANTED ->
                        WeatherContentUI(weatherState, onNavigateToTemperature)
                    else ->
                        PermissionRequestUi { composePermissionState.launchPermissionRequest() }
                }
```